### DISCUSSION TITLE
error compiling with firebasex plugin

### DISCUSSION THREAD

**Discussion by danielosky** [admob-plus/admob-plus#679] [Q&A]
Hi, I have a version conflict with the "cordova-plugin-firebasex" plugin. I have tried several ways to make the 2 plugins work together but I can't get the app to compile.

this is the error
```
> Task :app:processReleaseMainManifest FAILED
C:\apps\app\platforms\android\app\src\main\AndroidManifest.xml:32:13-58 Error:
        Attribute property#android.adservices.AD_SERVICES_CONFIG@resource value=(@xml/ga_ad_services_config) from [com.google.android.gms:play-services-measurement-api:21.6.2] AndroidManifest.xml:32:13-58
        is also present at [com.google.android.gms:play-services-ads-lite:23.0.0] AndroidManifest.xml:92:13-59 value=(@xml/gma_ad_services_config).
        Suggestion: add 'tools:replace="android:resource"' to <property> element at AndroidManifest.xml to override.

See https://developer.android.com/r/studio-ui/build/manifest-merger for more information about the manifest merger.
```

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
I have the same set-up without troubles.

1. **community-admob-plus-cordova** - `1.29.3`
2. **cordova-plugin-firebasex** -  `16.5.0`

My `config.xml` starts with;

`<?xml version="1.0" encoding="utf-8"?>
<widget id="foo.bar.app" version="1.2.3" xmlns="http://www.w3.org/ns/widgets" xmlns:android="http://schemas.android.com/apk/res/android" xmlns:cdv="http://cordova.apache.org/ns/1.0">`

And has this included;

` <platform name="android">
    <preference name="AndroidXEnabled" value="true" />`

**Reply by danielosky** [reply]
Thank you very much, I didn't know about the existence of "community-admob-plus-cordova"... Could you tell me how it differs from the original "admob-plus-cordova"?

**Reply by 1n3JgKl9pQ6cUMrW** [reply]
In short;

There is the original `admob-plus-cordova` made by Ratson.

But after some time development for this plug-in stopped and/ or slowed-down.

It had to be updated, because some (Google) libraries were sunset within a few months.

Since Ratson didn't have the time to do this, the community on Github decided to fork the original plug-in and updated the repository by itself; `community-admob-plus-cordova`.

So `community-admob-plus-cordova` was the exact same as the stable release for `admob-plus-cordova`, but with updated libraries.

After that fork, the beta of `admob-plus-cordova` went in development and is different from both the original `admob-plus-cordova` and `community-admob-plus-cordova`.

I use `community-admob-plus-cordova` for about 18 months now and it works perfect (banner / interstitial / native ad).

When `admob-plus-cordova` is out of beta, maybe I'll swith back to the "original".

**Reply by danielosky** [reply]
A question friend, when you open an interstitial and then when you close it "document.addEventListener('resume'')" I apply "location.reload();" and this makes me open a url "localhost/index.html" in the browser separately, do you know why it happens with this plugin community-admob-plus-cordova?

**Reply by 1n3JgKl9pQ6cUMrW** [reply]
I have never done that (reloading the app after closing the interstitial).

Maybe it has nothing to do with the plugin, and this is how `cordova` works?

I assume you are working with `cordova`.

You can also update the page without reloading it, e.g. by `ajax` or `location.href = location.href`.

But it goes out of the scope for this plugin.

**Reply by danielosky** [reply]
but with the original "admob-plus-cordova" in "resume" I had the "location.reload();" and it reloads well in the same app 

and with "community-admob-plus-cordova" the browser opens with "localhost/index.html"

**Reply by 1n3JgKl9pQ6cUMrW** [reply]
I don't know, in a regular `Cordova set-up` you never ever reload your `index.htm`.

`localhost/index.html` should be the only one, single html-file in your complete app.

If you want to reload / refresh or re-initialize the app, you do this by the default `AJAX architecture`.

Just set-up a routine that resets all content / variables / GUI, wrap it inside an `AJAX call` and away you go.

Or just call the native `Cordova function` for the app restart, you can also wrap it inside a plug-in;

https://github.com/huangang/cordova-plugin-restartapp

`navigator.app.restartApp();`

**Comment by ollm** [comment]
It seems to be the same issue as https://github.com/admob-plus/admob-plus/issues/672

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
I also have `cordova-android@13.0.0` but not the bèta-version of the AdMob-plugin.

So maybe it's linked to the bèta?

**Comment by hooliapps** [comment]
Hello, for info I use the beta version on Prod with 0 problems with UMP
Consent + Interstitial + Banner.



Le mar. 6 août 2024 à 19:04, 1n3JgKl9pQ6cUMrW ***@***.***> a
écrit :
